fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/directory/FDBDirectory.java

```diff
@@ -368,10 +407,11 @@ public void rename(@Nonnull final String source, @Nonnull final String dest) {
         LOGGER.trace("rename -> source={}, dest={}", source, dest);
         final byte[] key = metaSubspace.pack(source);
         CompletableFuture<byte[]> completableFuture;
-        completableFuture = context.ensureActive().get(key).exceptionally( (e) -> {
-            throw new RecordCoreArgumentException("Invalid source name in rename function for source: " + source);
-        });
-        completableFuture.thenAcceptAsync( (value) -> {
+        completableFuture = context.instrument(FDBStoreTimer.Events.LUCENE_RENAME,
+                context.ensureActive().get(key).exceptionally((e) -> {
```

Comment:
This should, again, be a `Wait`, also this is only timing the read of the `key`, not the actual rename.

I think you wanted:

```
context.asyncToSync(FDBStoreTimer.Waits.WAIT_LUCENE_RENAME,
    context.ensureActive().get(key).thenApply(value -> {
             this.fileReferenceCache.invalidate(source);
             context.ensureActive().set(metaSubspace.pack(dest), value);
             context.ensureActive().clear(key);
        }));
```